controller/array_action/array_mediator_ds8k.py

```diff
@@ -370,17 +374,144 @@ def unmap_volume(self, volume_id, host_name):
+        if not target_api_volume.flashcopy:
+            logger.error(
+                "FlashCopy relationship not found for target volume: {}".format(snapshot_name))
+            raise array_errors.SnapshotNameBelongsToVolumeError(target_api_volume.name,
+                                                                self.service_address)
+        fcrel = self._get_flashcopy(target_api_volume.flashcopy[0].id)
+        source_volume_name = self.get_volume_name(fcrel.source_volume.id)
+        return self._generate_snapshot_response(target_api_volume, source_volume_name)
+
+    def _create_similar_volume(self, target_volume_name, source_volume_name):
+        logger.info(
+            "creating target api volume '{0}' from source volume '{1}'".format(target_volume_name,
+                                                                               source_volume_name))
+        source_api_volume = self._get_api_volume_by_name(source_volume_name)
+        if not source_api_volume:
+            raise array_errors.VolumeNotFoundError(source_volume_name)
+        capabilities = {config.CAPABILITIES_SPACEEFFICIENCY: None}
+        size_in_bytes = int(source_api_volume.cap)
+        pool = source_api_volume.pool
+        return self.create_volume(target_volume_name, size_in_bytes, capabilities, pool)
+
+    def _create_flashcopy(self, source_volume, target_volume):
+        logger.info(
+            "creating FlashCopy relationship from '{0}' to '{1}'".format(source_volume.volume_name,
+                                                                         target_volume.volume_name))
+        source_volume_id = get_volume_id_from_scsi_identifier(source_volume.id)
+        target_volume_id = get_volume_id_from_scsi_identifier(target_volume.id)
+        try:
+            api_flashcopy = self.client.create_flashcopy(source_volume_id=source_volume_id,
+                                                         target_volume_id=target_volume_id,
+                                                         options=["persistent"])
+        except (exceptions.ClientError, exceptions.ClientException) as ex:
+            if ERROR_CODE_ALREADY_FLASHCOPY in str(ex.message).upper():
+                raise array_errors.SnapshotAlreadyExists(target_volume.volume_name,
+                                                         self.service_address)
+            elif ERROR_CODE_VOLUME_NOT_FOUND_OR_ALREADY_PART_OF_CS_RELATIONSHIP in str(
+                    ex.message).upper():
+                raise array_errors.VolumeNotFoundError('{} or {}'.format(source_volume_id,
+                                                                         target_volume_id))
+            else:
+                raise ex
+        if not self.validate_flashcopy(api_flashcopy.id):
+            # #TODO Delete flashcopy CSI-1340
+            logger.info("Flashcopy is not in a valid state")
+        return self._get_api_volume_by_id(target_volume_id)
+
+    @retry(Exception, tries=11, delay=1)
```

Comment:
after GA, this should be aligned according to the decision in [CSI-1505](https://github.com/IBM/ibm-block-csi-driver/pull/158#discussion_r433168025)